dashboard/app/models/script.rb

```diff
@@ -888,66 +888,80 @@ def has_older_version_progress?(user)
-      # Load custom scripts from Script DSL format
-      custom_files.map do |script|
-        name = File.basename(script, '.script')
-        base_name = Script.base_name(name)
-        name = "#{base_name}-#{new_suffix}" if new_suffix
-        script_data, i18n =
-          begin
-            ScriptDSL.parse_file(script, name)
-          rescue => e
-            raise e, "Error parsing script file #{script}: #{e}"
-          end
-
-        lesson_groups = script_data[:lesson_groups]
-        custom_i18n.deep_merge!(i18n)
-        # TODO: below is duplicated in update_text. and maybe can be refactored to pass script_data?
-        scripts_to_add << [{
-          id: script_data[:id],
-          name: name,
-          hidden: script_data[:hidden].nil? ? true : script_data[:hidden], # default true
-          login_required: script_data[:login_required].nil? ? false : script_data[:login_required], # default false
-          wrapup_video: script_data[:wrapup_video],
-          new_name: script_data[:new_name],
-          family_name: script_data[:family_name],
-          properties: Script.build_property_hash(script_data).merge(new_properties)
-        }, lesson_groups]
-      end
+    scripts_to_add = []
+
+    custom_i18n = {}
+    # Load custom scripts from Script DSL format
+    custom_files.map do |script|
+      name = File.basename(script, '.script')
+      base_name = Script.base_name(name)
+      name = "#{base_name}-#{new_suffix}" if new_suffix
+      script_data, i18n =
+        begin
+          ScriptDSL.parse_file(script, name)
+        rescue => e
+          raise e, "Error parsing script file #{script}: #{e}"
+        end
+
+      lesson_groups = script_data[:lesson_groups]
+      custom_i18n.deep_merge!(i18n)
+      # TODO: below is duplicated in update_text. and maybe can be refactored to pass script_data?
+      scripts_to_add << [{
+        id: script_data[:id],
+        name: name,
+        hidden: script_data[:hidden].nil? ? true : script_data[:hidden], # default true
+        login_required: script_data[:login_required].nil? ? false : script_data[:login_required], # default false
+        wrapup_video: script_data[:wrapup_video],
+        new_name: script_data[:new_name],
+        family_name: script_data[:family_name],
+        properties: Script.build_property_hash(script_data).merge(new_properties)
+      }, lesson_groups]
+    end
 
-      # Stable sort by ID then add each script, ensuring scripts with no ID end up at the end
-      added_scripts = scripts_to_add.sort_by.with_index {|args, idx| [args[0][:id] || Float::INFINITY, idx]}.map do |options, raw_lesson_groups|
-        add_script(options, raw_lesson_groups, new_suffix: new_suffix, editor_experiment: new_properties[:editor_experiment])
-      rescue => e
-        raise e, "Error adding script named '#{options[:name]}': #{e}", e.backtrace
+    # Print progress for dev environments and adhocs
+    debug = [:adhoc, :development].include?(rack_env)
```

Comment:
Rather than `rack_env`-dependent logic directly into the Script model, I'd prefer this be handled by configuration elsewhere. Would the existing `Rake.application.options.trace` option (which already adds extra debugging output to a Rake task) be sufficient for controlling this output? (e.g., `rake seed:scripts --trace`)

Reply:
That sounds like an improvement. I think ideally this wouldn't live in the Script model at all, but I don't want to go through with a bigger refactor now.

I do want to make sure that developers see this progress by default when following our setup instructions, rather than having to specify `--trace` explicitly, since I've heard a few people wondering if their install was frozen during seeding. But I can test that to make sure `--trace` is being passed through appropriately.

Reply:
I'll do this in a followup PR, since it's proving annoying to verify the behavior - `rake install` failing for other reasons it seems.

Reply:
`--trace` option sounds ideal. if it proves too time consuming, then a config variable on the CDO object which is set by default for development and adhoc seems like it could be an easier way to get the rack_env logic out of this model.